std/variant.d

```diff
@@ -1025,50 +1027,50 @@ public:
     //     return VariantN(lhs).opArithmetic!(VariantN, "-")(this);
     // }
     ///ditto
-    VariantN opMul(T)(T rhs) { return opArithmetic!(T, "*")(rhs); }
+    VariantN opBinary(string op : "*", T)(T rhs) { return opArithmetic!(T, "*")(rhs); }
     ///ditto
-    VariantN opDiv(T)(T rhs) { return opArithmetic!(T, "/")(rhs); }
+    VariantN opBinary(string op : "/", T)(T rhs) { return opArithmetic!(T, "/")(rhs); }
     // ///ditto
     // VariantN opDiv_r(T)(T lhs)
     // {
     //     return VariantN(lhs).opArithmetic!(VariantN, "/")(this);
     // }
     ///ditto
-    VariantN opMod(T)(T rhs) { return opArithmetic!(T, "%")(rhs); }
+    VariantN opBinary(string op : "%", T)(T rhs) { return opArithmetic!(T, "%")(rhs); }
     // ///ditto
     // VariantN opMod_r(T)(T lhs)
     // {
     //     return VariantN(lhs).opArithmetic!(VariantN, "%")(this);
     // }
     ///ditto
-    VariantN opAnd(T)(T rhs) { return opLogic!(T, "&")(rhs); }
+    VariantN opBinary(string op : "&", T)(T rhs) { return opLogic!(T, "&")(rhs); }
     ///ditto
-    VariantN opOr(T)(T rhs) { return opLogic!(T, "|")(rhs); }
+    VariantN opBinary(string op : "|", T)(T rhs) { return opLogic!(T, "|")(rhs); }
     ///ditto
-    VariantN opXor(T)(T rhs) { return opLogic!(T, "^")(rhs); }
+    VariantN opBinary(string op : "^", T)(T rhs) { return opLogic!(T, "^")(rhs); }
     ///ditto
-    VariantN opShl(T)(T rhs) { return opLogic!(T, "<<")(rhs); }
+    VariantN opBinary(string op : "<<", T)(T rhs) { return opLogic!(T, "<<")(rhs); }
     // ///ditto
     // VariantN opShl_r(T)(T lhs)
     // {
     //     return VariantN(lhs).opLogic!(VariantN, "<<")(this);
     // }
     ///ditto
-    VariantN opShr(T)(T rhs) { return opLogic!(T, ">>")(rhs); }
+    VariantN opBinary(string op : ">>", T)(T rhs) { return opLogic!(T, ">>")(rhs); }
     // ///ditto
     // VariantN opShr_r(T)(T lhs)
     // {
     //     return VariantN(lhs).opLogic!(VariantN, ">>")(this);
     // }
     ///ditto
-    VariantN opUShr(T)(T rhs) { return opLogic!(T, ">>>")(rhs); }
+    VariantN opBinary(string op : ">>>", T)(T rhs) { return opLogic!(T, ">>>")(rhs); }
```

Comment:
All these functions are nearly identical. The whole point of having the template was so you can easily implement multiple operators with one set of code. For example:

```D
VariantN opBinary(string op, T)(T rhs) if (op == "+" || op == "-" || op = "*" || op = "/")
{
    return opArithmetic!(T, op)(rhs);
}
```

In fact, I'd say get rid of `opArithmetic` and `opLogic`, and just move the implementations into an appropriate `opBinary`